### PR TITLE
sketch for sf-bay-area

### PR DESCRIPTION
**Pull request by tibbl35**
added necessary lines to fetch and load sf data to db for trips. working on stations.

with these changes, you can do the following:

```
dl_bikedata (city = 'sf', data_dir = data_dir)
bikedb <- file.path (data_dir, 'sfdb')
store_bikedata (data_dir = data_dir, city = 'sf', bikedb = bikedb)
index_bikedata_db (bikedb = bikedb)
```

but not the following:

```
tm <- bike_tripmat(bikedb = bikedb, city='sf')
```

where one gets the error message:

```
Error in rcpp_import_stn_df(bikedb, sdf, city = "sf") : 
  Unable to insert stations for sf
```

which seems to be thrown [here](https://github.com/ropensci/bikedata/blob/master/src/read-station-files.h#L278]) or [here](https://github.com/ropensci/bikedata/blob/master/src/read-station-files.h#L290)

debugging C++ is not something i've done before. 

i tried throwing zErrMsg instead. and it seemed to complain about some string escaping, maybe? i can't quite tell.

```
 Error in rcpp_import_stn_df(bikedb, sf_stns, "sf") : 
  near "Farrell": syntax error 
3.
stop(structure(list(message = "near \"Farrell\": syntax error", 
    call = rcpp_import_stn_df(bikedb, sf_stns, "sf"), cppstack = NULL), .Names = c("message", 
"call", "cppstack"), class = c("std::runtime_error", "C++Error", 
"error", "condition"))) 
2.
rcpp_import_stn_df(bikedb, sf_stns, "sf") at store-bikedata.R#158
1.
store_bikedata(data_dir = data_dir, city = "sf", bikedb = bikedb) 
```

so i tried throwing the records with O'Farrell out and then running again. 

sorry i couldn't be more helpful. i'll give it a try again sometime this week.

### CODE REVIEW

**Comment by mpadge** [comment]
Thanks so much @tibbl35! That's fantastic work! The code you did is pretty much there, the only real problem - which you already identified - is the stations. These data, as with other NABSA systems, include the station data in with the raw trip data files, so don't need a separate station data table to be read from anywhere. The station reading is instead linked in C++-world via the [`import_to_station_table` function](https://github.com/ropensci/bikedata/blob/master/src/read-station-files.h), using the `stationqry` set up during [`read_one_line_nabsa`](https://github.com/ropensci/bikedata/blob/master/src/read-city-files.h#L641-L642). The stations are then written following the reading of trip files right at the end of [`rcpp_import_to_trip_table`](https://github.com/ropensci/bikedata/blob/master/src/sqlite3db-add-data.h).

In short: All that seems to be missing is the first stage listed in the [current wiki](https://github.com/ropensci/bikedata/wiki/How-to-add-a-new-city) of modifying the `src/` files, and I think that, because these files *should be* structured identical to both `"ph"` and `"la"`, all you'll need to do is extend any mention of those to include `"sf"` as well. That should be only in [`src/sqlit3db-add-data.h`](https://github.com/ropensci/bikedata/blob/master/src/sqlite3db-add-data.h). You could just change that and re-open the same PR if you want.

Also please feel free to update the wiki if you thought any step was unclear, wrong, inaccurate, whatever. Thanks again for the sterling help here!

**Comment by tibbl35** [comment]
thanks @mpadge this is helpful ill give it a shot in the coming days.

**Comment by tibbl35** [comment]
@mpadge looks like the SF bay area data do not include 'trip_id', which philadelphia and LA do. 

i gather i need to change something in read_one_line_nabsa but not sure what. 

i'll try to take a look over the next few days but thought you or others might find the info useful anyway.

**Comment by tibbl35** [comment]
@mpadge really enjoying the comments, btw

**Comment by mpadge** [comment]
@tibbl35 FYI: Here's an all-in-one commit to add data for [Minneapolis/St Paul](https://github.com/ropensci/bikedata/commit/82acc28ad3a4f6a585066046429e42b909365d82)

**Comment by tibbl35** [comment]
@mpadge definitely still interested. i'll give it a shot in the next 2 days and we'll know one way or the other whether i can help by end of day tomorrow.
